src/vm-memory/src/mmap.rs

```diff
@@ -460,6 +460,63 @@ impl GuestMemoryMmap {
     pub fn is_dirty_tracking_enabled(&self) -> bool {
         self.regions.iter().all(|r| r.dirty_bitmap().is_some())
     }
+
+    pub fn read_from<F>(
+        &self,
+        addr: GuestAddress,
+        src: &mut F,
+        count: usize,
+    ) -> result::Result<usize, vm_memory_upstream::guest_memory::Error>
+    where
+        F: Read,
+    {
+        self.try_access(
+            count,
+            addr,
+            |_offset,
+             len,
+             caddr,
+             region|
+             -> result::Result<usize, vm_memory_upstream::guest_memory::Error> {
+                let page_size = match unsafe { libc::sysconf(libc::_SC_PAGESIZE) } {
+                    -1 => panic!("Failed to read: {}", errno::Error::last()),
+                    ps => ps as usize,
+                };
+                let len = std::cmp::min(len, page_size);
```

Comment:
Calling `libc::sysconf()` on each of these calls can be avoided by caching the page_size in a `lazy_static` or some other global const.

But looking closer I think page-size has nothing to do with this part anyway.

The enforced `len` of max 4096 is there to split a large read/write to multiple underlying calls where each call does at most 4096 bytes using a temporary buffer.

I believe this is intended to avoid temp buffer allocations of unbounded (or at least very big sizes).

For example when reading a 1GB kernel file to guest memory, we don't want to allocate 1GB temporary vec. Instead we read 4k bytes at a time.

So the old code `let len = std::cmp::min(len, 4096)` was correct.
But pls use a global const to define 4096 and add a comment around its usage logic.